solr/core/src/java/org/apache/solr/core/ConfigSetService.java

```diff
@@ -87,20 +78,59 @@ private static ConfigSetService instantiate(CoreContainer coreContainer) {
     }
   }
 
-  private static void bootstrapDefaultConfigSet(ConfigSetService configSetService) throws IOException {
-    if (configSetService.checkConfigExists("_default") == false) {
+  public void bootstrapConfigSet() {
+    // bootstrap _default conf, bootstrap_confdir and bootstrap_conf if provided via system property
+    String confDir = System.getProperty("bootstrap_confdir");
```

Comment:
It's better to declare variables as late as possible (just-in-time when needed), not at early as possible